### DISCUSSION TITLE
Feature Request: allow configuring blackout/privacy time blocks per camera

### DISCUSSION THREAD

**Discussion by JeanMertz** [koush/scrypted#1196] [Ideas]
In our family, we're using Scrypted with PoE cameras to secure our house's perimeter.

My SO asked me if it's possible to also have an indoor camera in our garage and potentially other indoor areas, but _only_ have them stream/record during nighttime, as to not invade our family's privacy during the day when we're up and about.

It doesn't appear such a feature exists in Scrypted. It does allow you to configure recording periods, but we're not using an NVR (yet), and instead feed the streams into HKSV with its record-on-motion feature. Additionally, we don't just want to black out recordings, but also live streams.

I could technically mess around with cutting the HKSV connection or just straight up disconnecting the camera using physical switches, but that gets messy and results in disconnection notifications etc., which I'd want to avoid.

Ideally, Scrypted would have a per-camera feature which allows you to configure "blackout time blocks". During these periods, the camera would only show a black screen when streaming (and thus also won't trigger a recording on motion), and the audio would also be muted during this period.

**Comment by koush** [comment]
This feature exists in Scrypted NVR, and I think you can set homekit to only record/stream when home or not home or something.

**Comment by JeanMertz** [comment]
> This feature exists in Scrypted NVR

Correct, but I'm not using Scrypted NVR (or any recording, other what HomeKit itself offers).

> and I think you can set homekit to only record/stream when home or not home or something.

That is correct, but I'm asking to block recording+stream **while people are home**, but at specific time blocks. In our case, we want to black out those cameras during the daytime, and only allow them to stream/record at night.

**Reply by koush** [reply]
This sounds like a feature request that should be sent to Apple.

**Reply by JeanMertz** [reply]
Well. I also expose the camera through Home Assistant, so ideally I would do this “at the source”. In which case it would be Scrypted, since otherwise it would have to be implemented by individual camera manufacturers, which is obviously never going to happen.

But is it fair to say that, given your pushback so far, you don’t think this would be a good fit for Scrypted? Would it at all be possible to do this as a plugin outside the core Scrypted code base?

**Reply by koush** [reply]
you'd need to fork the homekit (and maybe rebroadcast) plugins to selectively disable streams.

**Reply by JeanMertz** [reply]
Gotcha. So every plugin that distributes the stream to an external system would have to be modified? E.g. same for the Home Assistant plugin?

There is no way to have some kind of middleware plugin that captures the stream input, and modifies it (e.g. makes every pixel black) before the Homekit/HASS/... plugins get the feed?
